pkg/tsdb/elasticsearch/healthcheck.go

```diff
@@ -90,31 +88,125 @@ func (s *Service) CheckHealth(ctx context.Context, req *backend.CheckHealthReque
 		logger.Error("Error reading response body bytes", "error", err)
 		return &backend.CheckHealthResult{
 			Status:  backend.HealthStatusUnknown,
-			Message: "Failed to read response",
-		}, err
+			Message: "Health check failed: Failed to read response",
+		}, nil
 	}
 
 	jsonData := map[string]any{}
 
 	err = json.Unmarshal(body, &jsonData)
 	if err != nil {
-		logger.Error("Error during json unmarshal of the body", "error", err)
+		truncatedBody := string(body)
```

Comment:
Added to improve the error from this comment: https://github.com/grafana/grafana/issues/81767#issuecomment-2503435358

Since the body could be anything really, decided to limit it to 100 characters, but lmk if there's a better way

Reply:
I think instead of logging it, we can return it untruncated in the message instead. That way the user can at least see it in the frontend and doesn't have to dig in the logs for more info.

Reply:
I'm returning it in the message now; however, I still think it would be good to limit the characters. In case the user mistakenly enters the url to Kibana, for example, they'd get the html page in the response, which is probably too long to display. I increased the number of characters to 200 though, which might help with debugging. But lmk if you feel strongly about this!
<img width="1207" alt="Screenshot 2025-02-07 at 14 39 36" src="https://github.com/user-attachments/assets/0fd6326d-4d92-466b-8216-e08b4dea13b3" />